main.go

```diff
@@ -365,18 +365,39 @@ const indexSettings = `
                     "type":     "ngram",
                     "min_gram": 3,
                     "max_gram": 3
-                }
+				}
             },
             "analyzer": {
                 "trigrams": {
                     "type":      "custom",
                     "tokenizer": "standard",
-                    "filter":   [
+                    "filter": [
                         "lowercase",
                         "trigrams_filter"
                     ]
-                }
-            }
+				},
+				"locations": {
+					"tokenizer": "location_tokenizer",
+					"filter": [
+						"lowercase",
+						"word_delimiter"
+					]
```

Comment:
So this basically keeps/indexes only the tail part of the full location, so `Central District` in `USA > Washington > Seattle > Central District`. It also tokenizes that piece though, so that a simple match on `District` will match this, so a match_phrase should be used instead.